lib/mongo/retryable/read_worker.rb

```diff
@@ -218,7 +218,7 @@ def legacy_read_with_retry(session, server_selector, &block)
         
         if is_retryable_exception?(e)
           raise e if attempt > client.max_read_retries || session&.in_transaction?
-        elsif e.retryable? && !session&.in_transaction?
+        elsif e.respond_to?(:retryable?) && e.retryable? && !session&.in_transaction?
```

Comment:
should we want to match the new method [modern_read_with_retry](https://github.com/braze-inc/mongo-ruby-driver/blob/6322113530ca1619e5e4fee05135c5a7e2784231/lib/mongo/retryable/read_worker.rb#L198) way of handling these new errors instead? e.g. handle write_retryable errors here